packages/react-native/Libraries/Text/TextInput/RCTBackedTextInputDelegateAdapter.mm

```diff
@@ -267,7 +267,12 @@ - (void)textViewDidChange:(__unused UITextView *)textView
 - (void)textViewDidChangeSelection:(__unused UITextView *)textView
 {
   if (_lastStringStateWasUpdatedWith && ![_lastStringStateWasUpdatedWith isEqual:_backedTextInputView.attributedText]) {
```

Comment:
```suggestion
  if (_lastStringStateWasUpdatedWith && ![_lastStringStateWasUpdatedWith isEqual:_backedTextInputView.attributedText] && [self textView:_backedTextInputView shouldChangeTextInRange:NSMakeRange(0, _backedTextInputView.text.length) replacementText:_backedTextInputView.text]) {
```
I've reviewed the current implementation and I'm thinking we might streamline the conditional logic here. By consolidating the conditions into a single `if` statement instead of `nested if`. 

It retains all the original checks but in a more linear format.

Reply:
Don't know which is better :) , seems it's too long for the condition check.